### PR TITLE
Add support for PHP 8.4 (drop <7.1)

### PR DESCRIPTION
**Pull request by bnf**
This PR is an alternative to https://github.com/TYPO3/class-alias-loader/pull/32.
It drops support for PHP <7.1 in order to support PHP 8.4 deprecation free without requiring arguments to be removed or added.
(Changing arguments caused issues in previous changes, because old and new code is mixed during `composer update` withing one process)

### CODE REVIEW

**Comment by bnf** [comment]
Run in TYPO3 testing infrastructure: https://git.typo3.org/typo3/CI/cms/-/jobs/3846652

**Review by sbuerk — APPROVED** [review]

File: src/Config.php [hunk 1]
```diff
@@ -55,7 +55,8 @@ public function get($configKey)
             throw new \InvalidArgumentException('Configuration key must not be empty', 1444039407);
         }
         // Extract parts of the path
-        $configKey = str_getcsv($configKey, '.');
+        $configKey = str_getcsv($configKey, '.',  '"', '\\');
```
Comment (helhum) [inline src/Config.php:58]: how is this change related?

Comment (bnf) [reply src/Config.php:58]: See commit msg:

> Additional the call to str_getcsv is adapted to set all optional
arguments as the defaults will change in PHP >= 9 and PHP 8.4
deprecated the omission of the optional arguments.

**Review by lolli42 — APPROVED** [review]

File: src/Config.php (hunk 1 above)
Comment (helhum) [reply src/Config.php:58]: Thanks!

**Review by helhum — APPROVED** [review]
